src/index.js

```diff
@@ -53,57 +69,73 @@ export default source => {
   }
 
   // get a list of all function nodes
-  const functionNodes = flatten(ast).filter(propEquals('type', 'FunctionDeclaration'))
+  const contracts = ast.children.filter(child => child.type === 'ContractDefinition')
+const functionAndEventNodes = contracts.map(contract=>contract.subNodes).flat()
+    .filter(propEquals('type', ['FunctionDefinition', 'EventDefinition']))
 
   // analyze the security of the functions
-  const analyzedNodes = functionNodes.map(node => {
-    const functionCallees = callees(node).map(node => node.callee)
+  const analyzedNodes = functionAndEventNodes.map(node => {
+    const functionCallees = callees(node)
+      .map(statement => {
+        switch(statement.type ) {
+          case "EmitStatement": 
+            return statement.eventCall.expression.name  
+          case "ExpressionStatement": 
+            const expression = statement.expression.expression
+            if (expression.name) return statement.expression.expression.name
+            if (expression.type === 'MemberAccess') 
+              return expression.memberName
+        }
+      })
+
+    //CONSTRUCTOR
+    if (!node.name) {
+      handleSpecials(node)
+    }
+
     return {
       name: graphNodeName(node.name),
```

Comment:
I would recommend constructing the name like this:

```js
const name = graphNodeName(
  node.name || (node.isConstructor ? 'constructor' : node.isFallback ? 'fallback' : 'UNKNOWN')
)

return {
  name,
  ...
}
```

Reply:
done